homework/index.js

```diff
@@ -30,15 +30,106 @@
+          });
+          const repositoryForks = createAndAppend('div', repositoriesSection, {
+            class: 'repForks',
+          });
+          const repositoryDate = createAndAppend('div', repositoriesSection, { class: 'repDate' });
+          filteredData.forEach(allData => {
+            createAndAppend('span', repositoryName, { text: 'Repository Name:' });
+            createAndAppend('a', repositoryName, {
+              text: allData.name,
+              href: allData.html_url,
+              target: '_blank',
+            });
+            createAndAppend('span', repositoryDescription, { text: 'Description:' });
+            createAndAppend('p', repositoryDescription, { text: allData.description });
+            createAndAppend('span', repositoryForks, { text: 'Forks:' });
+            createAndAppend('p', repositoryForks, { text: allData.forks });
+            createAndAppend('span', repositoryDate, { text: 'Date:' });
+            createAndAppend('p', repositoryDate, { text: allData.updated_at });
+
+            const contributorSection = createAndAppend('div', container, {
+              class: 'ContInfo-side',
+            });
+            createAndAppend('h4', contributorSection, {
+              text: 'Contributions',
+              class: 'contributorHeader',
+            });
+            const contributorWrapper = createAndAppend('div', contributorSection, {
+              class: 'contributorWrapper',
+            });
+            return fetch(allData.contributors_url)
+              .then(contributorData => contributorData.json())
+              .then(contData => {
+                contData.forEach(contributor => {
+                  const contributorInfo = createAndAppend('a', contributorWrapper, {
+                    href: contributor.html_url,
+                    target: '_blank',
+                    class: 'contributorWrapper',
+                  });
+                  const contributorDiv = createAndAppend('div', contributorInfo, {
+                    class: 'contributor',
+                  });
+                  createAndAppend('img', contributorDiv, {
+                    class: 'contImage',
+                    src: contributor.avatar_url,
+                  });
+                  createAndAppend('h4', contributorDiv, {
+                    class: 'contName',
+                    text: contributor.login,
+                  });
+                  createAndAppend('span', contributorDiv, {
+                    class: 'contNumber',
+                    text: contributor.contributions,
+                  });
+                });
+              })
+              .catch(err => {
+                createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+              });
+          });
+        });
+    }
+    selectRepo.addEventListener('change', fetchSingleData);
+    fetchSingleData();
+  }
+
+  function main(url) {
+    fetchAllRepoData(url).then(() => getRepoData(url));
```

Comment:
Hi Ahmat - upon page load, I see that there are two network requests going for the `/repos` endpoint. This happens because you call the `fetch` endpoint twice, once from the `main()` and then again inside `getRepoData` by calling an internal function `fetchSingleData`. Can you please try to correct this behaviour?